src/core/Sample.cpp

```diff
@@ -117,58 +114,35 @@ auto Sample::operator=(Sample&& other) -> Sample&
 	return *this;
 }
 
-bool Sample::play(sampleFrame* dst, PlaybackState* state, int numFrames, float desiredFrequency, Loop loopMode)
+bool Sample::play(sampleFrame* dst, PlaybackState* state, size_t numFrames, float desiredFrequency, Loop loopMode) const
 {
-	if (numFrames <= 0 || desiredFrequency <= 0) { return false; }
+	const auto pastBounds = state->m_frameIndex >= m_endFrame || (state->m_frameIndex < 0 && state->m_backwards);
+	if (loopMode == Loop::Off && pastBounds) { return false; }
```

Comment:
Do we still want to check for `desiredFrequency <= 0` or is this handled elsewhere now?

Reply:
I feel like the code is simpler if we make the assumption that the caller gives us resonable parameters. If the caller gave us a `desiredFrequency` of `-440.0f` for example, that's on them, and the bug should be fixed at the call site. Otherwise, if we add this guard, the bug may never be found.